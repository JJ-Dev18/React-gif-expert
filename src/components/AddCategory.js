import React, { useState } from 'react'
import Proptypes from 'prop-types'


export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
   

      const handleInputChange = (e) => {
         console.log(e.target.value);
         setInputValue(e.target.value);

         console.log('HandleInputChange llamado')
         
     }

     const handleSubmit = (e) =>{
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }
     }
    return (
        <form onSubmit={handleSubmit}>
            <h2>Add Category</h2>
           
             <input 
             type="text"
             value={inputValue}
             onChange={ handleInputChange}
             ></input>
             
        </form>
    )

   
}

AddCategory.propTypes = {

    setCategories : Proptypes.func.isRequired

}



