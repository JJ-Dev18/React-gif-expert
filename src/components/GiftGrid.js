import React  from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

// import { getGifs } from '../Helpers/getGifs';


export const GiftGrid = ({category}) => {

    // const [images, setImages] = useState([])
    const {data: images ,loading} = useFetchGifs(category);
  
    // useEffect(() => {
    //    getGifs(category).then( imgs => setImages(imgs))
    // },[category])
   

    return (
       <>
         <h3 className="animate__animated animate__rubberBand">{category}</h3>
         {loading && <p className="animate__animated animate__flash"> Loading ...</p>}

        
         <div className="card-grid">  
               {
                  images.map(img =>(
                    <GifGridItem 
                    key={img.id}
                    {...img}/>
                  )) 
               }     
        </div>
       </>
        
    )
}

GiftGrid.propTypes = {

   category : PropTypes.string.isRequired
}