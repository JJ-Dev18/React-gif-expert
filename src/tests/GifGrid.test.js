import '@testing-library/jest-dom'
import React from 'react'
import { shallow } from 'enzyme'
import { GiftGrid } from '../components/GiftGrid'
import { useFetchGifs } from '../hooks/useFetchGifs'
jest.mock('../hooks/useFetchGifs');


describe('Prueba de el GifGrid', () => {
    
    test('Se debe cargar correctamente el GifGrid', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading : true 
        });
        const wrapper = shallow(<GiftGrid category ='one punch'/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFechGifs', () => {
        
        const gifs = [{
            id:'ABC',
            url: 'http:s//localhost/cualquier/cosa.jpg',
            title :'Cualquier cosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading : false 
        });
         
        const wrapper = shallow(<GiftGrid category ='one punch'/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    });

    
    
    
})
