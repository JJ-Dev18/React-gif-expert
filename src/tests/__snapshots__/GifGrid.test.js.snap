// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de el GifGrid Se debe cargar correctamente el GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rubberBand"
  >
    one punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba de el GifGrid debe de mostrar items cuando se cargan imagenes useFechGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rubberBand"
  >
    one punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http:s//localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
